### ISSUE TITLE
DNS addresses *.cluster.local cannot be resolved - total fail of all services

### ISSUE THREAD

**Issue by ChristofSchwarz** [ChristofSchwarz/qseok_on_Microk8s#1]
In my latest deployment of Microk8s I cannot resolve any cluster internal address like *.kubernetes.svc.cluster.local

Reproduce: this fails:
`kubectl exec -ti qlik-nats-0 -- nslookup kubernetes.default.svc.cluster.local`

This works:
`kubectl exec -ti qlik-nats-0 -- nslookup kubernetes.default.svc`

Subsequently, no service of qlik can communicate with any other, since the do using qualified service names like http://qlik-tenants.default.svc.cluster.local:8000 ...

**Comment by ChristofSchwarz** [comment]
**dig** is helpful with debugging. Get the **dnsutils** like this:
`kubectl apply -f https://k8s.io/examples/admin/dns/dnsutils.yaml`
then 
`kubectl exec -ti dnsutils -- dig kubernetes.default.svc.cluster.local`

**Comment by ChristofSchwarz** [comment]
Error Result looks like this:
`sudo microk8s kubectl exec -ti dnsutils -- nslookup kubernetes.default.svc.cluster.local`
```
Server:         10.152.183.10
Address:        10.152.183.10#53

** server can't find kubernetes.default.svc.cluster.local.3k32yaozxvnuteru2c5qb54rqg.bx.internal.cloudapp.net: SERVFAIL
```
good results look like this:
`sudo microk8s kubectl exec -ti dnsutils -- nslookup kubernetes.default.svc`
```
Server:         10.152.183.10
Address:        10.152.183.10#53

Name:   kubernetes.default.svc.cluster.local
Address: 10.152.183.1
```

**Comment by ChristofSchwarz** [comment]
Get current configuration of KubeDNS (coredns)
`kubectl describe configmap coredns -n kube-system`

Check the logs of **coredns**
`kubectl logs --selector k8s-app=kube-dns -n kube-system`

Restart the coredns pod:
`kubectl delete pod -n kube-system -l k8s-app=kube-dns`

**Comment by ChristofSchwarz** [comment]
I found the solution to my issue, and it was apparently a "stupid one".

* When I enabled DNS in my MicroK8s cluster, it sets the "forward" parameter of the Upstream DNS name servers to be 8.8.8.8 and 8.8.4.4 (two Google name servers).
 * Now, my Ubuntu VM runs on Azure, and the fully qualified name of a service it is something ending with *.internal.cloudapp.net. Such addresses are unknown to the Google DNS.
 * The fix simply was to change the "Corefile" (which is a configmap in kube-system and which is the configuration for CoreDNS) by adding an Azure DNS server like 168.63.129.16 in the "forward" setting before the 8.8.8.8 ...
 * check the steps here: https://github.com/ChristofSchwarz/qseok_on_Microk8s/blob/master/fix_coredns_azure.sh
